q2_phylogenomics/_assemble.py

```diff
@@ -118,25 +118,30 @@ def make_pileups(sorted: BAMFilesDirFmt,
     return result
 
 
-def consensus_sequence(pileups: PileUpFilesDirFmt
+def consensus_sequence(pileups: PileUpFilesDirFmt,
+                       min_depth: int = 10,
                        ) -> (pd.DataFrame, DNASequencesDirectoryFormat):
     features = DNASequencesDirectoryFormat()
     # what's the right way to get this file handle?
     sequences_fp = str(features.path / 'dna-sequences.fasta')
     table_data = {}
-    feature_ids = []
-    sample_ids = []
+    sample_ids = set()
+    feature_ids = set()
     with open(sequences_fp, 'w') as sequences_f, \
             tempfile.TemporaryDirectory() as tmpdirname:
         for path, view in pileups.pileups.iter_views(PileUpTSVFormat):
             ivar_cmd = ['ivar', 'consensus',
                         '-p', 'consensus-genome',
-                        '-n', 'N']
+                        '-n', 'N',
+                        '-m', str(min_depth)]
             run_command(ivar_cmd, cwd=tmpdirname,
                         stdin=open(str(pileups.path / path)))
 
             sample_id = path.stem
-            sample_ids.append(sample_id)
+            if sample_id in table_data:
```

Comment:
how else could we handle this? should/could we merge pileups? we want to avoid sample-id clashes, but what if we have e.g., tiled amplicon and shotgun data that we want to merge? Maybe that just gets merged after building a feature table.

I see where the replicate sample ids crept into the unit tests, and maybe it is not really a case that we would encounter in real life, but I guess I just wonder whether we should build in a safety valve just in case the need arises where, e.g., samples are sequenced in replicate.

Reply:
I think it would make the most sense to merge them at the feature table stage in the use case that you're describing. That way we're also encouraging the user to look at the different replicates and be explicit about how they want to handle the merge.